src/ConnectAuth.ts

```diff
@@ -0,0 +1,192 @@
+import { AuthError, AuthErrorCode } from './AuthError';
+import { decodeUnverifiedConnectJwt, verifyConnectJwt } from './Jwt';
+import { RequestReader } from './requestReader';
+import { ConnectJwt, ContextQsh, CredentialsLoader, InstallType } from './types';
+
+interface CommonVerifyArgs<E> {
+  baseUrl: string;
+  requestReader: RequestReader;
+  loadCredentials: CredentialsLoader<E>;
+  skipQueryStringHashCheck?: boolean; // Bitbucket does not implement Query String Hash
+}
+
+export type VerifyInstallArgs<E> = CommonVerifyArgs<E>;
+
+export interface VerifyInstallNewResponse {
+  type: InstallType.newInstall;
+  clientKey: string;
+}
+
+export interface VerifyInstallUpdateResponse<E> {
+  type: InstallType.update;
+  clientKey: string;
+  connectJwt: ConnectJwt;
+  storedEntity: E;
+}
+
+export type VerifyInstallResponse<E> = VerifyInstallNewResponse | VerifyInstallUpdateResponse<E>;
+
+export interface VerifyRequestArgs<E> extends CommonVerifyArgs<E> {
+  useContextJwt?: boolean;
+}
+
+export interface VerifyRequestResponse<E> {
+  connectJwt: ConnectJwt;
+  storedEntity: E;
+}
+
+export interface VerifyQueryStringHashArgs {
+  baseUrl: string;
+  connectJwt: ConnectJwt;
+  requestReader: RequestReader;
+  useContextJwt?: boolean;
+  skipQueryStringHashCheck?: boolean; // Bitbucket does not implement Query String Hash
+}
+
+/**
+ * Entry point for Connect request authentication.
+ */
+export class ConnectAuth {
+  /**
+   * Verifies a Connect request installation.
+   * Use this method to make sure the request is valid before persisting any data.
+   * This method handles both new installations and re-installations or installation updates.
+   */
+  static async verifyInstall<E>({
```

Comment:
Methods were renamed to the form of verb + noun.

Reply:
I've heard "install" as a noun a lot but it seems to be more like an IT jargon. Like https://wordtype.org/of/install and some other places have it but not old-good dictionaries. Will change that.